### PR TITLE
mongosh: refactor

### PR DESCRIPTION
**Pull request by wegank**
## Description of changes

https://github.com/NixOS/nixpkgs/pull/256997#issuecomment-1733022691

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build mongosh mongosh.passthru.tests

File: pkgs/development/tools/mongosh/default.nix [hunk 1]
```diff
@@ -1,37 +1,78 @@
...
+  # lerna ERR! ENOGIT The git binary was not found, or this is not a git repository.
   postPatch = ''
-    ln -s ${./package-lock.json} package-lock.json
+    if [ -x "$(command -v git)" ]; then
+      export HOME="$(mktemp -d)"
+      git config --global user.name "Nix Builder"
+      git config --global user.email "nix-builder@nixos.org"
+      git config --global init.defaultBranch "main"
+      git init .
+    fi
```
Comment (lilyinstarlight) [inline pkgs/development/tools/mongosh/default.nix:33]: Wait is this all that is needed to make lerna "work" here?

Comment (wegank) [reply pkgs/development/tools/mongosh/default.nix:33]: I'd also like to drop the constraint that the source must be a Git repo; I didn't manage to do that.

Comment (lilyinstarlight) [reply pkgs/development/tools/mongosh/default.nix:33]: I wonder where that happens. If you really got lerna to work with `buildNpmPackage`, though, with such a minor change, then it bodes well for other packages to be able to do the same without too much effort (I'd mentally written lerna off as a would-be-nice-to-have-in-the-future-but-not-priority thing due to pains with it before)

**Comment by wegank** [comment]
Lost interest...

**Pull request closed without merging**